File: Ejercicios/C#/Clases/Coche.cs
public class Coche
{
    public string Marca;
    public string Modelo;
    public int Año;

    public Coche()
    {
        Marca = "Genérico";
        Modelo = "Base";
        Año = 2000;
    }

    public Coche(string marca, string modelo, int año)
    {
        Marca = marca;
        Modelo = modelo;
        Año = año;
    }

    public Coche(string marca, string modelo)
    {
        Marca = marca;
        Modelo = modelo;
        Año = 2020;
    }

    public Coche(Coche otro)
    {
        Marca = otro.Marca;
        Modelo = otro.Modelo;
        Año = otro.Año;
    }
}

File: Ejercicios/Java/Clases/Coche.java
public class Coche {
    private String marca;
    private String modelo;
    private int anio;

    public Coche() {
        this("Genérico", "Base", 2000);
    }

    public Coche(String marca, String modelo, int anio) {
        this.marca = marca;
        this.modelo = modelo;
        this.anio = anio;
    }

    public Coche(String marca, String modelo) {
        this(marca, modelo, 2020);
    }

    public Coche(Coche otro) {
        this(otro.marca, otro.modelo, otro.anio);
    }
}
